docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2055 @@
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+The consensus reactor gossips consensus messages, by sending or forwarding them
+to peers.
+
+The `VoteSetMaj23` message is used by the consensus reactor to query peers
+regarding vote messages (prevotes or precommits) they have for a specific
+block.
+These queries are only triggered when +2/3 votes are observed.
+
+The value of `peer_query_maj23_sleep_duration` is the interval between sending
+those queries to a peer.
+
+## Storage
+In production environments, configuring storage parameters accurately is essential as it can greatly impact the amount
+of disk space utilized.
+
+CometBFT supports storage pruning to delete data indicated as not needed by the application or the data companion.
+Other than the pruning interval and compaction options, the configuration parameters in this section refer to the data
+companion. The applications pruning configuration is communicated to CometBFT via ABCI.
+
+Note that for some databases (GolevelDB), the data often does not get physically removed from storage due to the DB backend
+not triggering compaction. In these cases it is necessary to enable forced compaction and set the compaction interval accordingly.
+
+### storage.discard_abci_responses
+Discard ABCI responses from the state store, which can save a considerable amount of disk space.
+```toml
+discard_abci_responses = false
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `false` |
+|                     | `true`  |
+
+If set to `false` ABCI responses are maintained, if set to `true` ABCI responses will be pruned.
+
+ABCI responses are required for the `/block_results` RPC queries.
+
+### storage.experimental_db_key_layout
+
+The representation of keys in the database. The current representation of keys in Comet's stores is considered to be `v1`.
+
+Users can experiment with a different layout by setting this field to `v2`. Note that this is an experimental feature
+and switching back from `v2` to `v1` is not supported by CometBFT.
+
+If the database was initially created with `v1`, it is necessary to migrate the DB before switching to `v2`. The migration
+is not done automatically.
+
+```toml
+experimental_db_key_layout = 'v1'
+```
+
+| Value type          | string |
+|:--------------------|:-------|
+| **Possible values** | `v1`   |
+|                     | `v2`   |
+
+- `v1` - The legacy layout existing in Comet prior to v1.
+- `v2` - Order preserving representation ordering entries by height.
+
+If not specified, the default value `v1` will be used.
+
+### storage.compact
+
+If set to true, CometBFT will force compaction to happen for databases that support this feature and save on storage space.
+
+Setting this to true is most benefits when used in combination with pruning as it will physically delete the entries marked for deletion.
```

Comment:
```suggestion
Setting this to true is most beneficial when used in combination with pruning as it will physically delete the entries marked for deletion.
```